docs: Add dot shorthands guide to Flutter concepts (#13180)
flutter/website: src/content/ui/dot-shorthands.md

Here is a side-by-side comparison of building a simple `Container`:

### Without dot shorthands
```dart
Container(
  alignment: Alignment.center,
  padding: const EdgeInsets.all(16.0),
  child: Column(
    mainAxisAlignment: MainAxisAlignment.center,
    crossAxisAlignment: CrossAxisAlignment.start,
    children: [
      Text(
        'Hello World',
        style: TextStyle(
          fontWeight: FontWeight.bold,
        ),
      ),
    ],
  ),
);
```

### With dot shorthands
```dart
Container(
  alignment: .center, // Instead of Alignment.center,
  padding: const .all(16.0), // Instead of EdgeInsets.all(16.0)
  child: Column(
    mainAxisAlignment: .center, // Instead of MainAxisAlignment.center
    crossAxisAlignment: .start, // Instead of CrossAxisAlignment.start
    children: [
      Text(
        'Hello World',
        style: TextStyle(
          fontWeight: .bold, // Instead of FontWeight.bold
        ),
      ),
    ],
  ),
);
```

## Where to use dot shorthands

Dot shorthands work anywhere the Dart compiler has a clear "context type",
meaning it knows exactly what type it expects. In Flutter, this is almost

> everywhere inside a widget's property list. 

Reviewer: > In Flutter, this is almost everywhere inside a widget's property list.

For some reason that I can't fully identify, I don't love this sentence. It feels like it's potentially overpromising? I much prefer the following paragraph which bubbles up concrete common targets. Maybe consider cutting this sentence, but up to you :)

Reviewer 2: Yeah, it doesn't feel like this sentence enhances. I'd also delete.

The most common targets for dot shorthands in Flutter are:

*   **Enums**: `MainAxisAlignment`, `CrossAxisAlignment`, `BoxFit`, `TextDirection`.
*   **Static properties and methods**: `FontWeight` (constants like `.bold`). 
*   **Constructors**: `EdgeInsets.all()`, `BorderRadius.circular()`.

### Example: enums

> When a property expects an `enum`, such as `mainAxisAlignment`, you can omit the

Reviewer: I wonder if we should make this more broad since this can be used for more than properties/arguments/etc. Maybe consider:

```suggestion
When the context type is an `enum`, such as `MainAxisAlignment`, you can omit the
```

enum's name and just provide the value preceded by a dot (`.`):

```dart
Row(
  mainAxisAlignment: .spaceEvenly, // Infers MainAxisAlignment.spaceEvenly
  children: [ /* ... */ ],
)
```

### Example: static properties

Static properties work when the context type is exactly the class that defines the property. A common example is text styling with `FontWeight`:

```dart
Text(
  'Feature highlights',
  style: TextStyle(
    fontWeight: .bold, // Infers FontWeight.bold
  ),
)
```

### Example: constructors

You can also use dot shorthands for named constructors. Many Flutter layout properties accept a base class like `EdgeInsetsGeometry`. To support dot shorthands, Flutter adds redirecting constructors to these base classes that point to the appropriate subclasses.

```dart
Padding(
  padding: .symmetric(horizontal: 16.0, vertical: 8.0), // Infers EdgeInsetsGeometry.symmetric
  child: Text('Spaced out text'),
)
```

You can even use `.new` to call an unnamed constructor, though this is less

> common in standard widget trees:

Reviewer: This feels a bit opinionated (though it does seem to be a common opinion). But since these are official docs, maybe consider being objective:

```suggestion
You can also use `.new` to call an unnamed constructor:
```
